### PR TITLE
Refactor Settings to inherit from QSettings

### PR DESCRIPTION
**Pull request by Adrok24**
*Add typing

*Add test cases for this class

### CODE REVIEW

File: SentinelHub/settings.py [hunk 1]
```diff
@@ -65,43 +66,38 @@ class Settings:
     _CREDENTIAL_STORE_PARAMETERS = {"base_url", "client_id", "client_secret"}
     _STORE_PARAMETERS = _AUTO_SAVE_STORE_PARAMETERS | _CREDENTIAL_STORE_PARAMETERS
 
-    def __init__(self):
+    def __init__(self, path=None):
```
Comment (zigaLuksic) [inline SentinelHub/settings.py:69]: annotation missing?

File: SentinelHub/settings.py [hunk 2]
```diff
@@ -65,43 +66,38 @@ class Settings:
...
+        super().__init__(path)
         self.load_local_settings()
         self._auto_save = True
 
-    def __setattr__(self, key, value):
+    def __setattr__(self, key: str, value: Any) -> None:
         """Whenever one of the attributes from _AUTO_SAVE_STORE_PARAMETERS is set it is automatically saved to
         QGIS store
         """
-        if self._auto_save and key in self._AUTO_SAVE_STORE_PARAMETERS:
```
Comment (zigaLuksic) [inline SentinelHub/settings.py:76]: does the `_auto_save` attribute do anything after this change?

File: SentinelHub/settings.py [hunk 3]
```diff
@@ -65,43 +66,38 @@ class Settings:
...
             store_path = self._get_store_path(parameter)
-            qsettings.setValue(store_path, getattr(self, parameter))
-
-    def _get_store_path(self, parameter_name):
-        """Provides a location of the parameter in the local store"""
-        return f"{self._STORE_NAMESPACE}/{parameter_name}"
+            self.setValue(store_path, getattr(self, parameter))
 
-    def copy(self):
+    def copy(self) -> None:
```
Comment (zigaLuksic) [inline SentinelHub/settings.py:101]: wrong type annotation

File: SentinelHub/settings.py [hunk 4]
```diff
@@ -61,47 +63,43 @@ class Settings:
...
-    def _get_store_path(self, parameter_name):
-        """Provides a location of the parameter in the local store"""
-        return f"{self._STORE_NAMESPACE}/{parameter_name}"
-
-    def copy(self):
-        """Provides a copy of a Settings object instance"""
-        return copy.copy(self)
+            self.setValue(store_path, getattr(self, parameter))
+
+    def copy(self) -> T:
```
Comment (zigaLuksic) [inline SentinelHub/settings.py:100]: Here the type should be `Settings` due to L101. I don't think you should lose sleep over this, unlikely someone will subclass it further.

Comment (Adrok24) [reply SentinelHub/settings.py]: fixed

**Pull request merged**